test/integration/udp_echo_integration_test.cc

```diff
@@ -0,0 +1,85 @@
+#include "common/network/listen_socket_impl.h"
+
+#include "test/integration/integration.h"
+#include "test/integration/utility.h"
+
+namespace Envoy {
+
+std::string echo_config;
+
+class UdpEchoIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
+                               public BaseIntegrationTest {
+public:
+  UdpEchoIntegrationTest() : BaseIntegrationTest(GetParam(), echo_config) {}
+
+  // Called once by the gtest framework before any EchoIntegrationTests are run.
+  static void SetUpTestSuite() {
+    echo_config = ConfigHelper::BASE_UDP_LISTENER_CONFIG + R"EOF(
+    listener_filters:
+      name: envoy.listener.udpecho
+    filter_chains:
```

Comment:
filter_chains for udp listener does not make sense currently. However, removing this fails proto validation as this is minimum required = 1. Can we relax the proto validation and do the validation in code for only tcp filters that are expected to have filter_chains

Reply:
Yes I think that's fine, but please make sure you have an explicit test for that case.